From 96158ba05d9e28bb7e4802c5426b7d8f2f2322b8 Mon Sep 17 00:00:00 2001
From: Rico Ganahl <rico.ganahl@bytesatwork.ch>
Date: Mon, 6 Feb 2023 07:17:16 +0100
Subject: [PATCH] bytesatwork: bawconfig: add 4 GB as 3 GB RAM

Fix 4 GB RAM equipped SoM on non-quad IMX8M.
According to the imx8mm datasheet only Quad-A53 supports
more than 3 GB RAM. Therefore define dedicated enum to
use 4 GB as 3 GB RAM.

Signed-off-by: Rico Ganahl <rico.ganahl@bytesatwork.ch>
(cherry picked from commit 1d017f3668dee1e30281da79b2f8db2f80da1662)
---
 board/bytesatwork/baw_config/baw_config.c | 2 ++
 board/bytesatwork/baw_config/baw_config.h | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/board/bytesatwork/baw_config/baw_config.c b/board/bytesatwork/baw_config/baw_config.c
index 4dfa4fd520..f5e229cea7 100644
--- a/board/bytesatwork/baw_config/baw_config.c
+++ b/board/bytesatwork/baw_config/baw_config.c
@@ -30,6 +30,8 @@ const struct baw_config_name_pair baw_config_ram_name[] = {
 	{ M6_RAM_MT53D512M32D2DS_053, "MT53D512M32D2DS-053 2 GB LPDDR4 @ 1500 MHz" },
 	{ M6_RAM_MT53E768M32D4DT_053, "MT53E768M32D4DT-053 3 GB LPDDR4 @ 1500 MHz" },
 	{ M6_RAM_MT53D1024M32D4DT_053, "MT53D1024M32D4DT-053 4 GB LPDDR4 @ 1500 MHz" },
+	{ M6_RAM_MT53D1024M32D4DT_053_3GB,
+		"MT53D1024M32D4DT-053 4 GB using 3 GB @ 1500 MHz" },
 	{ -1, NULL },
 };
 
diff --git a/board/bytesatwork/baw_config/baw_config.h b/board/bytesatwork/baw_config/baw_config.h
index fefdfd5405..d689fa1a97 100644
--- a/board/bytesatwork/baw_config/baw_config.h
+++ b/board/bytesatwork/baw_config/baw_config.h
@@ -33,6 +33,10 @@ typedef enum {
 	M6_RAM_MT53D512M32D2DS_053 = 9,
 	M6_RAM_MT53E768M32D4DT_053 = 10,
 	M6_RAM_MT53D1024M32D4DT_053 = 11,
+	/* According to the imx8mm datasheet only Quad-A53 supports more than
+	 * 3 GB RAM. Therefore define dedicated enum to use 4 GB as 3 GB RAM.
+	 */
+	M6_RAM_MT53D1024M32D4DT_053_3GB = 12,
 } baw_config_ram_t;
 
 typedef enum {
-- 
2.39.2

